main.py

```diff
@@ -192,53 +206,89 @@ def add_workout_to_schedule(self,schedule,workout,duration):
         return self.post_req('schedulestep',data)
 
 
-    def delete_workout(self, id):
+    def delete_workout(self, id=None):
         workouts = self.get_req('workout')
         list_of_id = list()
         for workout in workouts['results']:
```

Comment:
Use workouts.get('results', []) in order for the 'for' loop to iterate over an empty list(and implicitly have no effect) if there is no 'results' key in 'workouts'